### PR TITLE
add validations and associated model tests for bills

### PR DESCRIPTION
**Pull request by jrpespisa**

### CODE REVIEW

**Comment by jrpespisa** [comment]
Validations complete! Ready for review and merge..

File: app/models/bill.rb [hunk 1]
```diff
@@ -2,4 +2,15 @@ class Bill < ActiveRecord::Base
...
   accepts_nested_attributes_for :unit
-end
\ No newline at end of file
+
+  validate :bill_received_is_date?
+  validates :bill_received, presence: true
+  validates :amount, presence: true, numericality: true, inclusion: {in: [0, 9999]}
+
+  def bill_received_is_date?
+    if !bill_received.is_a?(Date)
```
Comment (mxie) [inline app/models/bill.rb]: How about `unless` instead of `if !...`?

**Comment by mxie** [comment]
Changes look pretty straightforward. Just had 1 suggestion around syntax. Also, could you squash some of the commits and update the commit messages as appropriate?

**Comment by jrpespisa** [comment]
All set, I made the changes you suggested.

**Pull request merged**